Source/VCGeneration/Split.cs

```diff
@@ -868,6 +1036,65 @@ public Counterexample ToCounterexample(ProverContext context)
         return newBlocks;
       }
 
+      private static List<Block> PostProcess(List<Block> blocks)
+      {
+        void DeleteFalseGotos (Block b)
+        {
+          bool isAssumeFalse (Cmd c) { return c is AssumeCmd ac && ac.Expr is LiteralExpr le && !le.asBool; }
+          var firstFalseIdx = b.Cmds.FindIndex(c => isAssumeFalse(c));
+          if (firstFalseIdx != -1)
+          {
+            b.Cmds = b.Cmds.Take(firstFalseIdx + 1).ToList();
+            b.TransferCmd = (b.TransferCmd is GotoCmd) ? new ReturnCmd(b.tok) : b.TransferCmd;
+          }
+        }
+
+        void DeleteSubSumptions (Block b)
+        {
+          bool isAssumeSubsumption (Cmd c) { return c is AssumeCmd ac && QKeyValue.FindBoolAttribute(ac.Attributes, "subsumption"); }
+          b.Cmds = b.Cmds.Where(c => !isAssumeSubsumption(c)).ToList();
+        }
+
+        bool ContainsAssert(Block b)
+        {
+          bool isNonTrivialAssert (Cmd c) { return c is AssertCmd ac && !(ac.Expr is LiteralExpr le && le.asBool); }
+          return b.Cmds.Aggregate(false, (containsAssert, c) => containsAssert || isNonTrivialAssert(c));
+        }
+
+        // return blocks;
+        blocks.ForEach(b => DeleteSubSumptions(b));
+        blocks.ForEach(b => DeleteFalseGotos(b)); // make blocks ending in assume false leaves of the CFG-DAG -- this is probably unnecessary, may have been done previously
```

Comment:
Might there be non-assert blocks that `goto` a block ending in `assume false`? If so, it would be nice to propagate the `assume false` to predecessor blocks. (Perhaps this is done elsewhere.)